src/data/set/basic.lean

```diff
@@ -1765,16 +1760,27 @@ For the corresponding result for `subtype`, see `subtype.subsingleton`. -/
 instance subsingleton_coe_of_subsingleton [subsingleton α] {s : set α} : subsingleton s :=
 by { rw [s.subsingleton_coe], exact subsingleton_of_subsingleton }
 
+theorem subsingleton_anti {α : Type*} {s t : set α} (hst : s ⊆ t) (hs : subsingleton t) :
```

Comment:
I don't see why this is useful. `set.subsingleton s` should be the only spelling we care about.

Reply:
Well, we have `nontrivial_mono`, even before my recent work, and this is the analogous result.

Reply:
I have updated the description with more justification.

Reply:
Essentially: I will leave this in, unless you also think we should not have `nontrivial_mono`, in which case I'll remove that. But my feeling is we should have both or neither.

Reply:
We should also not have that. It's only used once in [`lie_module.nontrivial_max_triv_of_is_nilpotent`](https://leanprover-community.github.io/mathlib_docs/algebra/lie/nilpotent.html#lie_module.nontrivial_max_triv_of_is_nilpotent), and it's used to prove a lemma that should have been stated in terms of `set.nontrivial` instead. But removing it should be a separate PR. I can take care of that.

Reply:
I'd prefer if you not added `subsingleton_mono`, and we can remove `nontrivial_mono` later.

Reply:
Brilliant - I will do this.